### ISSUE TITLE
Automate masking threshold

### ISSUE THREAD

**Issue by astewartau** [QSMxT/QSMxT#37]
Apply the 'Background removal' masking step based on a gaussian graph.

Details here: https://www.nature.com/articles/s41598-021-02722-0 

Alternatively look at 3dClipLevel https://afni.nimh.nih.gov/pub/dist/doc/htmldoc/programs/3dClipLevel_sphx.html

How well does this work for abdominal imaging?

**Comment by iishiishii** [comment]
As the proposed method is not effective for t1w and the threshold is very inconsistent across scan, I'm exploring another histogram based algorithm: https://www.sciencedirect.com/science/article/pii/S0010482512000157

**Comment by astewartau** [comment]
Thanks for the work on this - this seems to work great! 

I made a few minor changes before integrating it. Importantly, the `op_string` generated by the threshold interface no longer includes the `-dilM` step after the `-ero` step. I removed the `-dilM` step because it seemed to introduce artefacts in the gold seed phantom data that I tested it on. This is likely because the automated gaussian mask is much more generous about what it includes than what we usually use, and so keeping the small amount of erosion is beneficial. This may require some further tweaking as we test more datasets. 

In the near future, we should also combine the `gaussian-based` masking option with `magnitude-based`. We can change the default `--threshold` setting for `magnitude-based` to `None`; in such cases, the automated thresholding should kick in.
